resteasy-core-spi/src/main/java/org/jboss/resteasy/spi/AsyncOutputStream.java

```diff
@@ -8,6 +10,13 @@
  */
 public abstract class AsyncOutputStream extends OutputStream {
 
+    public static int ASYNC_MESSAGE_WRITE_BUFFER_SIZE = ConfigurationFactory.getInstance()
+            .getConfiguration()
+            .getOptionalValue(
+                    "resteasy.async.output.stream.write.buffer.size",
+                    Integer.class
+            ).orElse(2048);
+
```

Comment:
I don't think this should be static. It can't be changed without, typically, restarting the JVM. The value could come from something like a deployment descriptor too which would not work well if there are multiple deployments.

Reply:
@jamezp is there any other better option to host this configuration ?

Reply:
I'm trying to think of how this could be done. It might be best to make it an instance variable with a simple getter. It would only be looked up when the a new instance of an `AsyncOutputStream` is created.

Reply:
The `ChunkOutputStream` is created for every response. If the fetching the value from the `Configuration` should be avoided on the `ProviderHelper#asyncWrite` method, in my understanding(considering the asyncWrite also will be called just once) having an instance value should also have a similar cost. Please correct me if I am wrong.

Reply:
That does seem correct to me now that I have a closer look. Sorry, I'm still familiarizing myself with this :) 

My only concern is this could impact performance to save the cost of the memory overhead.

Reply:
Actually we are trying to increase the performance with a slight memory overhead. 

As you know, we are working on a bug in the reactor-netty adapter [here](https://github.com/resteasy/Resteasy/pull/2882) where we had to write and flush each chunk(which is of size 2048 due to the ProviderHelper#asyncWrite implementation). This works fine for small data sizes. When we deal with larger data sizes, instead of flushing on every 2048 bytes(which may impact the performance), the application can choose to configure a higher size for this buffer so that the frequency of write and flush shall be minimised to get a better performance with the cost of a slight memory overhead.

Reply:
Ah okay got it. What we might need to do, and not in this change, is look for a way to keep the `Configuration` associated with the context of "deployment" rather than look it up each time which is where the expense is. We can file another JIRA for that though.

However, for this specific change we should do the lookup instead of using a static variable which is not friendly in multi-tenant environments.

Reply:
I understand that you would like to go back to adding this configuration lookup in the ProviderHelper#asyncWrite implementation. But based on your previous comment on this lookup being expensive, we are concerned on how big this impact would be on the performance provided the fact that this code will be executed for every request.

Reply:
Yes that is a concern of mine as well. However, it can't be static for containers like WildFly. Possibly next week I could look at a solution to better cache a configuration, but that won't do much good now. 

The expense is in the lookup of the `ConfigurationFactory` as it uses a `ServiceLoader` for the lookup.

Reply:
The only thing really blocking this is having the value static. We can't have that and really the fact that it's not final and public is a bit of a problem too.